File: api/Dtos/Achievement.cs
using api.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Dtos
{
    public class AchievementResponseDto
    {
        public Guid AchievementId { get; set; }
        public string? Name { get; set; }
        public int Page { get; set; } = 1;
        public int Xp { get; set; } = 1;
        public Guid? PartId { get; set; }
        public Part? Part { get; set; }
    }

    public class AchievementCreateDto : AchievementManipulate
    {

    }

    public class AchievementUpdateDto : AchievementManipulate
    {

    }

    public class AchievementManipulate
    {
        [Required, MaxLength(100)]
        public string? Name { get; set; }
        public int Page { get; set; } = 1;
        public int Xp { get; set; } = 1;
        [Required]
        public Guid? PartId { get; set; }
    }
}

File: web/src/dtos/achievement.ts
import { PartResponseDto } from "./part";
import { PlayerAchievementResponseDto } from "./Player";

export interface AchievementResponseDto {
  achievementId?: string;
  name?: string;
  page?: number;
  xp?: number;
  partId?: string;
  part?: PartResponseDto;
  playerAchievements?: PlayerAchievementResponseDto[];
}

export class AchievementUpdateDto {
  constructor(partId?: string) {
    this.partId = partId;
  }
  name?: string = "";
  page?: number = 1;
  xp?: number = 25;
  partId?: string = "";
}

